tests/components/test_canary.py

```diff
@@ -0,0 +1,108 @@
+"""The tests for the Canary component."""
+import unittest
+from unittest.mock import patch
+
+import homeassistant.components.canary as canary
+from canary.api import Location, Mode
+from tests.common import (
+    get_test_home_assistant)
+
+MODES_BY_NAME = {
+    "home": Mode({"id": 1, "name": "Home", "resource_uri": "/v1/home"}),
```

Comment:
I think you should mock the api classes/instances instead, so we don't need to add the library to the test requirements.